fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ReferencesRdfContext.java

```diff
@@ -48,21 +57,36 @@ public ReferencesRdfContext(final FedoraResource resource,
         throws RepositoryException {
         super(resource, idTranslator);
         property2triple = new PropertyToTriple(resource.getNode().getSession(), idTranslator);
-        concat(putStrongReferencePropertiesIntoContext());
-        concat(putWeakReferencePropertiesIntoContext());
+        session = resource.getNode().getSession();
+        putReferencesIntoContext(resource.getNode().getWeakReferences());
+        putReferencesIntoContext(resource.getNode().getReferences());
     }
 
-    private Iterator<Triple> putWeakReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getWeakReferences();
+    private void putReferencesIntoContext(final Iterator<Property> properties) throws RepositoryException {
+        while (properties.hasNext()) {
+            final Property p = properties.next();
+            concat(property2triple.apply(p));
 
-        return Iterators.concat(Iterators.transform(properties, property2triple));
+            for ( final PropertyIterator it = p.getParent().getProperties(); it.hasNext(); ) {
+                final Property potentialProxy = it.nextProperty();
+                if (potentialProxy.isMultiple()) {
+                    for ( Value v : potentialProxy.getValues() ) {
+                        putProxyReferencesIntoContext(v);
+                    }
+                } else {
+                    putProxyReferencesIntoContext(potentialProxy.getValue());
+                }
+            }
+        }
     }
-
-    private Iterator<Triple> putStrongReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getReferences();
-
-        return Iterators.concat(Iterators.transform(properties, property2triple));
-
+    private void putProxyReferencesIntoContext(final Value v) throws RepositoryException {
+        if (v.getType() == PATH) {
```

Comment:
This could help:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/converters/ValueConverter.java#L78
...as well as possibly this:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java#L232

Although when dealing with "references" on Fedora, we also need to be conscious of these:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java#L211 and
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java#L220